<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QDialog" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>693</width>
    <height>709</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>SerialPort Assistant</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res/mainwindow.qrc">
    <normaloff>:/logo.png</normaloff>:/logo.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="topLayout" stretch="0">
   <property name="spacing">
    <number>11</number>
   </property>
   <property name="leftMargin">
    <number>11</number>
   </property>
   <property name="topMargin">
    <number>11</number>
   </property>
   <property name="rightMargin">
    <number>11</number>
   </property>
   <property name="bottomMargin">
    <number>11</number>
   </property>
   <item>
    <widget class="QWidget" name="centralWidget" native="true">
     <layout class="QVBoxLayout" name="mainLayout">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>120</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tabServer">
         <attribute name="title">
          <string>服务器</string>
         </attribute>
         <widget class="QComboBox" name="comboBox">
          <property name="geometry">
           <rect>
            <x>220</x>
            <y>20</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_2">
          <property name="geometry">
           <rect>
            <x>220</x>
            <y>50</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_3">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>50</y>
            <width>121</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_4">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>20</y>
            <width>121</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>20</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>TCP地址</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>50</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>UDP地址</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>190</x>
            <y>20</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>190</x>
            <y>50</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton">
          <property name="geometry">
           <rect>
            <x>290</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>TCP侦听</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_2">
          <property name="geometry">
           <rect>
            <x>290</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>UDP侦听</string>
          </property>
         </widget>
         <widget class="QListView" name="listView">
          <property name="geometry">
           <rect>
            <x>350</x>
            <y>20</y>
            <width>256</width>
            <height>61</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_3">
          <property name="geometry">
           <rect>
            <x>610</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>全选</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_4">
          <property name="geometry">
           <rect>
            <x>610</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>断开</string>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="tabForward">
         <attribute name="title">
          <string>转发器</string>
         </attribute>
         <widget class="QComboBox" name="comboBox_5">
          <property name="geometry">
           <rect>
            <x>210</x>
            <y>50</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_5">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>SRC地址</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_6">
          <property name="geometry">
           <rect>
            <x>210</x>
            <y>20</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_6">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>启动</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_7">
          <property name="geometry">
           <rect>
            <x>600</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>全选</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_6">
          <property name="geometry">
           <rect>
            <x>180</x>
            <y>50</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_7">
          <property name="geometry">
           <rect>
            <x>50</x>
            <y>20</y>
            <width>121</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_8">
          <property name="geometry">
           <rect>
            <x>50</x>
            <y>50</y>
            <width>121</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_7">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>50</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>DST地址</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_8">
          <property name="geometry">
           <rect>
            <x>600</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>断开</string>
          </property>
         </widget>
         <widget class="QListView" name="listView_2">
          <property name="geometry">
           <rect>
            <x>340</x>
            <y>20</y>
            <width>256</width>
            <height>61</height>
           </rect>
          </property>
         </widget>
         <widget class="QLabel" name="label_8">
          <property name="geometry">
           <rect>
            <x>180</x>
            <y>20</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_9">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="tabClient">
         <attribute name="title">
          <string>客户端</string>
         </attribute>
         <widget class="QComboBox" name="comboBox_19">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>20</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_20">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>20</y>
            <width>151</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_21">
          <property name="geometry">
           <rect>
            <x>380</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_17">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>20</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>服务器地址</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_18">
          <property name="geometry">
           <rect>
            <x>250</x>
            <y>20</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_15">
          <property name="geometry">
           <rect>
            <x>450</x>
            <y>20</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>启动</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_20">
          <property name="geometry">
           <rect>
            <x>250</x>
            <y>50</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>端口</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_23">
          <property name="geometry">
           <rect>
            <x>380</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_16">
          <property name="geometry">
           <rect>
            <x>450</x>
            <y>50</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="text">
           <string>启动</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_21">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>50</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>服务器地址</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_24">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>50</y>
            <width>151</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_25">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>50</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="tabSerialPort">
         <attribute name="title">
          <string>串口</string>
         </attribute>
         <widget class="QLed" name="led" native="true">
          <property name="geometry">
           <rect>
            <x>470</x>
            <y>10</y>
            <width>24</width>
            <height>24</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="openCloseButton">
          <property name="geometry">
           <rect>
            <x>510</x>
            <y>10</y>
            <width>79</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>开始</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_22">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>10</y>
            <width>151</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_19">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>10</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>串口设备</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_26">
          <property name="geometry">
           <rect>
            <x>390</x>
            <y>40</y>
            <width>51</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_22">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>40</y>
            <width>50</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>服务器地址</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_27">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>40</y>
            <width>151</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_28">
          <property name="geometry">
           <rect>
            <x>290</x>
            <y>40</y>
            <width>65</width>
            <height>24</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="clearBox_2">
          <property name="geometry">
           <rect>
            <x>530</x>
            <y>40</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>转发</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QCheckBox" name="clearBox">
          <property name="text">
           <string>自动清空</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="hexSendBox">
          <property name="text">
           <string>HEX Send</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="loopBox">
          <property name="text">
           <string>循环发送</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="cycleBox">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="minimum">
           <number>-1</number>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="value">
           <number>60</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="sendButton">
          <property name="text">
           <string>发送</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="sendEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCheckBox" name="wordWrapBox">
          <property name="text">
           <string>Word Wrap</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="hexShowBox">
          <property name="text">
           <string>HEX Show</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="sendButton_2">
          <property name="text">
           <string>清空</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="recvEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QLed</class>
   <extends>QWidget</extends>
   <header>qled.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/mainwindow.qrc"/>
 </resources>
 <connections/>
</ui>
